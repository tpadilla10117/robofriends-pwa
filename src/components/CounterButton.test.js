import { shallow, mount, render } from 'enzyme';
import React from 'react';
import CounterButton from './CounterButton';

describe('CounterButton.js component:', () => {
    it('expect to render CounterButton component', () => {
        const mockColor = 'red';
        expect.assertions(1);
        expect(shallow(<CounterButton color={mockColor} />)).toMatchSnapshot();
    })

    it('correctly increments the counter', () => {
        const mockColor = 'red';
        const wrapper = shallow(<CounterButton color={mockColor} />);
        expect.assertions(2);

    /* Searching for an id by simulating a click event: */
        wrapper.find('[id="counter"]').simulate('click')
        expect(wrapper.state()).toEqual({ count: 2});

    /* Checking the props: */
        expect(wrapper.props().color).toEqual('red');
    })
})