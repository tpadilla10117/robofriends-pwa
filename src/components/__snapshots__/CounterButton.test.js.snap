// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterButton.js component: expect to render CounterButton component 1`] = `ShallowWrapper {}`;
